tests/test_models/test_inference/test_forecast.py

```diff
@@ -532,6 +533,196 @@ def test_forecast_in_sample_suffix_failed_not_implemented_in_sample(self, model,
         _test_prediction_in_sample_suffix(example_tsds, model, transforms, method_name="forecast", num_skip_points=50)
 
 
+class TestForecastOutSample:
+    """Test forecast on of future dataset.
+
+    Expected that NaNs are filled after prediction.
+    """
+
+    @staticmethod
+    def _test_forecast_out_sample(ts, model, transforms, prediction_size=5):
+        # fitting
+        ts.fit_transform(transforms)
+        model.fit(ts)
+
+        # forecasting
+        forecast_ts = ts.make_future(future_steps=prediction_size, tail_steps=model.context_size, transforms=transforms)
+        forecast_ts = make_forecast(model=model, ts=forecast_ts, prediction_size=prediction_size)
+
+        # checking
+        forecast_df = forecast_ts.to_pandas(flatten=True)
+        assert not np.any(forecast_df["target"].isna())
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (CatBoostPerSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
```

Comment:
I should probably merge master into current `unaligned_data` to add here `DeepARNativeModel`.